### PR TITLE
doc: tutorial updates for 3.0

### PR DESCRIPTION
**Pull request by jhamman**
This is a work in progress. Running through the tutorial helped identify a number of issues that we likely want to address before we release 3.0.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -155,7 +155,7 @@ argument accepted by all array creation functions. For example::
     >>> from numcodecs import Blosc
     >>> compressor = Blosc(cname='zstd', clevel=3, shuffle=Blosc.BITSHUFFLE)
     >>> data = np.arange(100000000, dtype='i4').reshape(10000, 10000)
-    >>> z = zarr.array(data, chunks=(1000, 1000), compressor=compressor)
+    >>> z = zarr.array(data, chunks=(1000, 1000), compressor=compressor, zarr_format=2)
```
Comment (jhamman) [inline docs/tutorial.rst:158]: This example works for `zarr_format=2` but requires https://github.com/zarr-developers/zarr-python/issues/2398 to work for `zarr_format=3`

File: docs/tutorial.rst [hunk 2]
```diff
@@ -300,32 +300,30 @@ Groups have a similar API to the Group class from `h5py
 
 Groups can also contain arrays, e.g.::
 
-    >>> z1 = bar.zeros('baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
+    >>> z1 = bar.zeros(name='baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```
Comment (jhamman) [inline docs/tutorial.rst:303]:
```suggestion
    >>> z1 = bar.zeros('baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```

see https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 3]
```diff
@@ -300,32 +300,30 @@ Groups have a similar API to the Group class from `h5py
...
-    <zarr.Array '/foo/bar/baz' (10000, 10000) int32>
+     <Array memory://4640612800/foo/bar/baz shape=(10000, 10000) dtype=int32>
 
-Arrays are known as "datasets" in HDF5 terminology. For compatibility with h5py,
-Zarr groups also implement the ``create_dataset()`` and ``require_dataset()``
-methods, e.g.::
+Arrays can also be created with the ``create_array()`` and ``require_array()`` methods, e.g.::
 
-    >>> z = bar.create_dataset('quux', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
+    >>> z = bar.create_array(name='quux', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```
Comment (jhamman) [inline docs/tutorial.rst:309]:
```suggestion
    >>> z = bar.create_array('quux', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```
see https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 4]
```diff
@@ -335,16 +333,16 @@ representation of the hierarchy, e.g.::
...
+The :func:`zarr.api.asynchronous.open` function provides a convenient way to create or
 re-open a group stored in a directory on the file-system, with sub-groups stored in
 sub-directories, e.g.::
 
     >>> root = zarr.open('data/group.zarr', mode='w')
     >>> root
-    <zarr.hierarchy.Group '/'>
-    >>> z = root.zeros('foo/bar/baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
+    <Group file://data/group.zarr>
+    >>> z = root.zeros(name='foo/bar/baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```
Comment (jhamman) [inline docs/tutorial.rst:343]:
```suggestion
    >>> z = root.zeros('foo/bar/baz', shape=(10000, 10000), chunks=(1000, 1000), dtype='i4')
```
https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 5]
```diff
@@ -362,9 +360,9 @@ property. E.g.::
 
     >>> root = zarr.group()
     >>> foo = root.create_group('foo')
-    >>> bar = foo.zeros('bar', shape=1000000, chunks=100000, dtype='i8')
+    >>> bar = foo.zeros(name='bar', shape=1000000, chunks=100000, dtype='i8')
```
Comment (jhamman) [inline docs/tutorial.rst:363]:
```suggestion
    >>> bar = foo.zeros('bar', shape=1000000, chunks=100000, dtype='i8')
```

https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 6]
```diff
@@ -362,9 +360,9 @@ property. E.g.::
...
     >>> bar[:] = 42
-    >>> baz = foo.zeros('baz', shape=(1000, 1000), chunks=(100, 100), dtype='f4')
+    >>> baz = foo.zeros(name='baz', shape=(1000, 1000), chunks=(100, 100), dtype='f4')
```
Comment (jhamman) [inline docs/tutorial.rst:365]:
```suggestion
    >>> baz = foo.zeros('baz', shape=(1000, 1000), chunks=(100, 100), dtype='f4')
```

https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 7]
```diff
@@ -440,7 +438,7 @@ storing application-specific metadata. For example::
 
     >>> root = zarr.group()
     >>> root.attrs['foo'] = 'bar'
-    >>> z = root.zeros('zzz', shape=(10000, 10000))
+    >>> z = root.zeros(name='zzz', shape=(10000, 10000))
```
Comment (jhamman) [inline docs/tutorial.rst:441]:
```suggestion
    >>> z = root.zeros('zzz', shape=(10000, 10000))
```

https://github.com/zarr-developers/zarr-python/issues/2402

File: docs/tutorial.rst [hunk 8]
```diff
@@ -1655,9 +1642,9 @@ Datetimes and timedeltas
 NumPy's ``datetime64`` ('M8') and ``timedelta64`` ('m8') dtypes are supported for Zarr
 arrays, as long as the units are specified. E.g.::
 
-    >>> z = zarr.array(['2007-07-13', '2006-01-13', '2010-08-13'], dtype='M8[D]')
+    >>> z = zarr.array(['2007-07-13', '2006-01-13', '2010-08-13'], dtype='M8[D]', zarr_format=2)
```
Comment (jhamman) [inline docs/tutorial.rst:1645]: Add note above explaining that datetimes/timedeltas are only supported by v2 data format for now.

**Review by jhamman — COMMENTED** [review]
self review 👇

**Comment by dstansby** [comment]
Woop woop! In case you don't know about it already, https://github.com/scientific-python/pytest-doctestplus?tab=readme-ov-file#fixing-existing-docstrings might be helpful

**Comment by jhamman** [comment]
closing in favor of #2589

**Pull request closed without merging**